javatests/arcs/core/storage/StorageProxyTest.kt

```diff
@@ -155,22 +159,62 @@ class StorageProxyTest {
         ).isTrue()
 
         val syncDeferred = CompletableDeferred<Boolean>()
-        storageProxy.addOnSync("testHandle") {
-            syncDeferred.complete(true)
+        coroutineScope {
+            storageProxy.addOnSync("testHandle") {
+                syncDeferred.complete(true)
+            }
         }
+        syncDeferred.complete(false)
         assertThat(syncDeferred.await()).isEqualTo((true))
     }
 
     @Test
-    fun getOnDesyncCalledWhenAdded() = runBlockingTest {
+    fun getOnSyncNotCalledWhenAddedIfNotSynced() = runBlocking {
+        val storageProxy = StorageProxy(mockStorageEndpointProvider, mockCrdtModel)
+
+        val syncDeferred = CompletableDeferred<Boolean>()
+        coroutineScope {
```

Comment:
This does the trick: the block passed to `coroutineScope` won't exit until all child jobs have completed.